### DISCUSSION TITLE
Semester 4 Woche 3

### DISCUSSION THREAD

**Discussion by TimSchnur** [Christian-2003/CrInGE#42] [Student Exchange]
Hallo zusammen,

nach reichlicher Überlegung und einem 👀 Review des vorhandenen Codes haben wir uns dazu entschlossen kein Refactoring vorzunehmen.
Dies liegt nicht zuletzt an unseren von Beginn des Projekt festgelegten 📑 Regeln, welche einen sauberen Code Stil begünstigten.
Des weiteren werden viele der Funktionen in den nächsten Wochen noch weiter 📈 wachsen und sich neuen Anforderungen anpassen müssen, wodurch sie ohnehin 🚧 überarbeitet werden.

Falls euch trotzdem eine unsaubere Stelle auffallen sollte, dürft ihr uns gerne 💬 darauf aufmerksam machen.

Aufgrund der kürzlichen Nachklausur und den baldigen Abgaben für AdA haben wir auch sonst nur 📉  wenige Fortschritte in dieser Woche erzielen können. Dieser erhöhte Aufwands durch AdA wird auch nächste Woche unser halbes Team beschäftigen, weshalb eine große Neuerung dort nicht zu erwarten ist.
Allerdings geht es danach in gewohnter 💪 Stärke weiter.

Wir werden euch natürlich trotzdem nächste Woche von unseren Ergebnissen berichten.

**Comment by PetrusQn** [comment]
Hallo Cringe-Team,
euer vorhandener Code scheint eine klare Form bzw. einen guten Stil zu haben - was sehr lobenswert ist. Daher ist es verständlich, dass ihr euch, für jetzt, gegen ein Refactoring entschieden habt. Der angesprochene, langsame, Fortschritt aktuell ist ebenfalls verständlich. Achtet nur darauf, dass euer Risiko "Unrealistische Zeit- und Aufwandsabschätzung" nicht gepaart mit einem verzögerten Zeitplan auftritt 😉 
Ich freue mich darauf nächste Woche wieder von euch zu hören 😃👍

**Comment by lukpr0** [comment]
Hallo Cringe-Team,
euer Code sieht für mich auf den ersten Blick ganz gut aus. Das einzige, was mir aufgefallen wäre, man könnte die ActionListener separat, anstatt innheralb der Funktion, in der sie gesetzt werden definieren, um die Länge einzelner Funktionen kürzer zu halten.
